segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -1038,6 +1040,26 @@ public void readTableEntriesDelta(WireCommands.ReadTableEntriesDelta readTableEn
 
     }
 
+    private void truncateIndexSegment(String segment, long offset) {
+        String indexSegment = getIndexSegmentName(segment);
+        try {
+            long indexSegmentOffset = IndexRequestProcessor.locateOffsetForIndexSegment(segmentStore, segment, offset, false);
+            if (indexSegmentOffset == 0) {
+                return;
+            }
+            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT)
+                    .whenComplete((v, exp) -> {
+                        if (exp != null) {
+                            log.warn("Truncation of index segment {} at offset {} can not be performed.", indexSegment, indexSegmentOffset);
+                        } else {
+                            log.info("Truncation index segment {} at offset {}.", indexSegment, indexSegmentOffset);
```

Comment:
Log message can be improved. "Successfully truncated index segment {} at offset {}."

Reply:
Done